tests/queries/0_stateless/01173_transaction_control_queries.sql

```diff
@@ -6,8 +6,8 @@ drop table if exists mt2;
 create table mt1 (n Int64) engine=MergeTree order by n;
 create table mt2 (n Int64) engine=MergeTree order by n;
 
-commit; -- { serverError INVALID_TRANSACTION } -- no transaction
-rollback; -- { serverError INVALID_TRANSACTION }
+commit;
```

Comment:
It should apply only to the MySQL compatibility. I don't think this change is correct. We can ask @tavplubix

Reply:
I prefer the strict version (and for `rollback` as well)

Reply:
The issue is that this way it breaks MySQL tools as cited in #57598 so that Clickhouse can't be used as a drop in replacement for MySQL.
I share your opinion that the strict version makes sense but this denies easy usage in such scenarios.

Is there a way to get the information that the query is coming from MySQL port from the session context? If so this behaviour could be set only for such queries. Woudl that be a satisfying solution for you?

Reply:
Yes, it makes sense, see `Context::getClientInfo()` and `ClientInfo::Interface`

Reply:
@tavplubix 

> Yes, it makes sense, see `Context::getClientInfo()` and `ClientInfo::Interface`

Thank you for your indications. I have changed the code based on ClientInfo::Interface to show this behaviour only for MySQL interfaces. I reverted back the test and created a new one for the MySQL client to verify the new behaviour.

Is there anything else I need to do to make this pull request mergeable?

CI seems ok as the 6 tests that did not work are no related to my changes.

Thank you!